pages/people/[id]/details.tsx

```diff
@@ -16,19 +16,24 @@ interface Props {
 const PersonAllocationsPage = ({ person }: Props): React.ReactElement => {
   const [showAddedCaseStatusConfirmation, setshowAddedCaseStatusConfirmation] =
     useState(false);
+  const [caseStatusConfirmationText, setCaseStatusConfirmationText] = useState(
+    'Flagged status added'
+  );
+
   const router = useRouter();
 
   useEffect(() => {
-    setshowAddedCaseStatusConfirmation(
-      Boolean(router.query.flaggedStatus) && person.contextFlag === 'C'
-    );
+    setshowAddedCaseStatusConfirmation(Boolean(router.query.flaggedStatus));
+    if (router.query.message) {
+      setCaseStatusConfirmationText(String(router.query.message));
+    }
   });
 
   return (
     <>
       {showAddedCaseStatusConfirmation && (
         <ConditionalFeature name="case-status">
-          <ConfirmationBanner title={'Flagged status added'} />
+          <ConfirmationBanner title={caseStatusConfirmationText} />
```

Comment:
Would this work instead

```
<ConfirmationBanner
            title={String(router.query.message) ?? 'Flagged status added'}
          />
```

Then we can remove a state from the component and reduce the useEffect

Reply:
I also use an assertion instead of creating a new string when possible, this saves us actually having to create the new string object.